File: SharpMap.Presentation.Web.SharpLayers/Layers/Wms/WmsLayerComponent.js
Type.registerNamespace('SharpMap.Presentation.Web.SharpLayers.Layers.Wms');

SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent = function(element) {
    SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent.initializeBase(this, [element]);

}
SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent.prototype = {
    initialize: function() {
        this._builderDelegate = Function.createDelegate(this, this._layerBuilderDelegate);
        SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent.callBaseMethod(this, 'initialize');
    },

    dispose: function() {
        SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent.callBaseMethod(this, 'dispose');
    },
    _layerBuilderDelegate: function() {
        var options = this.get_builderParams();
        var url = options.params.url;
        delete options.params.url;
        var params = options.params;
        delete options.params;
        return new OpenLayers.Layer.WMS(this.get_name(), url, params, options);
    }


}
SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent.registerClass('SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent', SharpMap.Presentation.Web.SharpLayers.Layers.LayerComponent);

File: SharpMap.Presentation.Web.SharpLayers/Layers/Wms/WmsLayerComponent.cs
using System.Web.UI;
using AjaxControlToolkit;

[assembly: WebResource("SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent.js", "text/javascript")]

namespace SharpMap.Presentation.Web.SharpLayers.Layers.Wms
{
    [ClientScriptResource("SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent",
        "SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent.js")]
    [TargetControlType(typeof (Control))]
    public class WmsLayerComponent : LayerComponent<WmsLayerBuilderParams>
    {
        protected override string ScriptComponentName
        {
            get { return "SharpMap.Presentation.Web.SharpLayers.Layers.Wms.WmsLayerComponent"; }
        }
    }
}